packages/core/src/tracing/opentelemetry-instrumentations/index.js

```diff
@@ -0,0 +1,11 @@
+/*
+ * (c) Copyright IBM Corp. 2023
+ */
+
+'use strict';
+
+module.exports = {
+  get otelInstrumentations() {
```

Comment:
_question:_ OTel has different policies around support for older Node.js versions than we do.

https://github.com/open-telemetry/opentelemetry-js/tree/main#supported-runtimes

> Only Node.js Active or Maintenance LTS versions are supported. Previous versions of node may work, but they are not tested by OpenTelemetry and they are not guaranteed to work.

I think it might be a good idea to avoid the risk of OTel modules causing issues for customers on older Node.js versions by disabling OTel integration on our end for older versions. (probably everything below 14?)

Not sure where we would want to do that, either here or we could automatically set the config flag for OTel integration to off on older versions, or put the condition into `packages/core/src/tracing/index.js`.

Reply:
If the node version is unsupported, we set `automaticTracingEnabled` to false in `normalizeConfig`.
And we only enable Otel if `automaticTracingEnabled` is true. So I guess we are safe here?

Reply:
Hm, not sure that is exactly what I mean. We set have this as our expression for supported versions: `semver.satisfies(version, '^10.4.0 || ^11 || >=12.0.0');` So we basically support Node.js 10.x and above.

The OTel project on the other hand only test their code against Node.js 14 and above on their CI. 

I was thinking we might want to explicitly disable this new OTel integration for Node.js 10 and 12, and only enable it for Node.js 14 and above. Customers on Node 10 and 12 can still use our "own" auto-tracing instrumentations (which we test extensively on Node 10 and 12 on CI), but we do not load the OTel code automatically on these old Node.js versions.

The reason is mainly that by doing that we would have the stronger guarantees from the OTel test suite/CI apply to this new integration as well. 

Of course, you added a lot of tests for the new OTel integration to our test suite as well. You have a point if you say that this makes it safe to run the OTel code as part of `@instana/collector` on Node.js 10 and 12 as well.

But: At the moment we depend on the OTel-related libraries with a version range -- `"@opentelemetry/api": "^1.4.0"` etc., all are referred to with a caret (`^`) in the version.

So one possible scenario would be this: Currently everything works on Node 10, and the version of let say `@opentelemetry/instrumentation-fs` we have in our `package-lock.json` (`0.7.1`) is compatible with Node.js 10. Then `@opentelemetry/instrumentation-fs` introduces some code that breaks on Node.js 10 and releases a new version 0.8.0. Their package.json file says `engine: >= 14`, so it wouldn't even be a breaking change from their perspective. Plus, they don't run tests on Node.js 10, so they wouldn't notice. Our tests would also not go red immediately because we still have version 0.7.1 in our package-lock.json. But someone doing a fresh install of `@instana/collector` would receive `@opentelemetry/instrumentation-fs@0.8.0` and it would break their app. 

So at a minimum I think we need to do either of those two thing (or both):
* Reference all the new packages via an exact version, not a version range.
* Disable the new OTel integration code specifically on all versions below Node.js 14.

Reply:
> Disable the new OTel integration code specifically on all versions below Node.js 14.

Great idea 👍 Let's do.